### PR TITLE
GA preview client library for .NET

### PR DESCRIPTION
**Pull request by dpokluda**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Review by chunyu3 — COMMENTED** [review]
The samples are not follow the .NET guideline. [https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-repository](https://nam06.safelinks.protection.outlook.com/?url=https%3A%2F%2Fazure.github.io%2Fazure-sdk%2Fdotnet_introduction.html%23dotnet-repository&data=05%7C01%7CYu.Chun%40microsoft.com%7C36ec088822a74e81f20308da49e5e239%7C72f988bf86f141af91ab2d7cd011db47%7C1%7C0%7C637903548951042270%7CUnknown%7CTWFpbGZsb3d8eyJWIjoiMC4wLjAwMDAiLCJQIjoiV2luMzIiLCJBTiI6Ik1haWwiLCJXVCI6Mn0%3D%7C3000%7C%7C%7C&sdata=ZgyDLOatMpCy3CO%2BMf2yMFmlKG8sZonMkAoIOlC5C4M%3D&reserved=0)

**Review by chunyu3 — CHANGES_REQUESTED** [review]
- change endpoint to be a Uri instead of a string.
- The samples are not follow the .NET guideline. [https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-repository]

**Comment by chunyu3** [comment]
Hi @dpokluda Please refer to [package requirements](https://github.com/Azure/azure-sdk-for-net/blob/main/doc/DataPlaneCodeGeneration/AzureSDKCodeGeneration_DataPlane_Quickstart.md#add-package-ship-requirements) to write samples, snippets and readme.

You also can follow our [PackageStarter](https://github.com/Azure/azure-sdk-for-net/blob/main/doc/DataPlaneCodeGeneration/AzureSDKCodeGeneration_DataPlane_Quickstart.md) to re-gen the .NET package which will comply all .Net requirements, then update the tests and samples.

**Comment by dpokluda** [comment]
I have removed the old samples and added new samples (first batch). I will add more samples later on.

**Comment by dpokluda** [comment]
For the account endpoint still being `string`. @lmazuel, I do already have the parameter marked (see [EndpointParameter](https://github.com/Azure/azure-rest-api-specs/blob/4124b7c2773a714303299f0cfd742b0d26d3bb5d/specification/deviceupdate/data-plane/Microsoft.DeviceUpdate/preview/2022-07-01-preview/deviceupdate.json#L4335)):
```
      "format": "uri",
      "type": "string",
```

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Sample1_HelloWorld.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+```C# Snippet:AzDeviceUpdateSample1_CreateDeviceUpdateClient
+var credentials = new DefaultAzureCredential();
+var client = new DeviceUpdateClient(accountEndpoint, instance, credentials);
+```
+
+## Enumerate all device update providers
+
+First, let's try to enumerate all available (already imported) device update providers.
+
+```C# Snippet:AzDeviceUpdateSample1_EnumerateProviders
```
Comment (archerzz) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Sample1_HelloWorld.md:20]: Where is this snippet from? I could not find it in test codes.

**Review by archerzz — CHANGES_REQUESTED** [review]
Please double check the snippets. They should be from test codes. See the guides below:
- https://github.com/Azure/azure-sdk-for-net/blob/main/doc/DataPlaneCodeGeneration/AzureSDKCodeGeneration_DataPlane_Quickstart.md#samples
- https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#updating-sample-snippets

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Sample1_HelloWorld.md (hunk 1 above)
Comment (dpokluda) [reply sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Sample1_HelloWorld.md]: I misunderstood how snippets work. I have removed the references to snippets. I would like to merge it now without them and then add the proper snippet references in the next version. Is that possible?

Comment (archerzz) [reply sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Sample1_HelloWorld.md]: That's OK

**Review by archerzz — APPROVED** [review]

**Comment by archerzz** [comment]
@dpokluda approved. please resolve the conflicts. thanks.

**Comment by dpokluda** [comment]
Should be resolved now. Since the conflict was in generated files, I have simply replaced those with the newly generated versions. Please let me know if anything else is needed. Thanks.

**Comment by archerzz** [comment]
@dpokluda Can you check the sample docs again? It seems there are still snippet names remaining: https://dev.azure.com/azure-sdk/public/_build/results?buildId=1690158&view=logs&j=b70e5e73-bbb6-5567-0939-8415943fadb9&t=2102385d-609d-5572-64d2-932661c7902f&l=18

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Sample1_HelloWorld.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
+# Basic device update enumerations
```
Comment (chunyu3) [inline sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Sample1_HelloWorld.md:1]: The samples do not comply our guideline. Please refer to https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-samples and https://github.com/Azure/azure-sdk-for-net/blob/main/doc/DataPlaneCodeGeneration/Create_Samples_Guidance.md to update the samples. and You can refer to the SDK package template https://github.com/Azure/azure-sdk-for-net/tree/main/sdk/template/Azure.Template

Comment (dpokluda) [reply sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Sample1_HelloWorld.md:1]: Can we ship this preview with these samples and address the feedback in our next PR?

**Comment by dpokluda** [comment]
> * change endpoint to be a Uri instead of a string.
> * The samples are not follow the .NET guideline. [https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-repository]

This has been addressed already.

File: sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Sample1_HelloWorld.md (hunk 2 above)
Comment (chunyu3) [reply sdk/deviceupdate/Azure.IoT.DeviceUpdate/samples/Sample1_HelloWorld.md:1]: As we discussed in the email, that for preview, we can ship this without compliance samples.

**Review by chunyu3 — APPROVED** [review]

**Pull request merged**